## 45% of end-end tests have passed
```
Total executed tests: 11
Failed tests: 6
Expected to fail tests: 0
Passed tests: 5
```
```
IDs of failed tests: 727231,741555,703133,740490,741612,741554 
```
<h3>Failed tests (6)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/keycard-see-recovery-phrase-again&PR_ID=22009&APK_NAME=22009.apk&TR_CASE_IDS=727231,741555,703133,740490,741612,741554">Rerun failed tests</a></li><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777477">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><p><blockquote><div># STEP: Adding new regular account</div><div>Device 1: Find `Button` by `accessibility id`: `add-account` </div></blockquote></p>

```
critical/test_wallet.py:552: in test_wallet_add_remove_regular_account
    self.wallet_view.add_regular_account(account_name=new_account_name)
../views/wallet_view.py:215: in add_regular_account
    self.add_account_button.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `add-account` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777480">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td><p><blockquote><div>Device 1: Could not reach Button by pressing system back button </div><div>Device 1: Find `Button` by `xpath`: `//android.view.ViewGroup[contains(@content-desc,'Account 1')]` </div></blockquote></p>

```
critical/test_wallet.py:349: in test_wallet_swap_flow_mainnet
    self.wallet_view.get_account_element().click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//android.view.ViewGroup[contains(@content-desc,'Account 1')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777478">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `network-dropdown` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/test_wallet.py:264: in test_wallet_balance_mainnet
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: For the Ether the wrong value 0.0362 is shown, expected 0.0262 in total
E    Device 1: For the Ether the wrong value 0.015 is shown, expected 0.005 on Mainnet
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777481">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td><p><blockquote><div>Device 1: Could not reach Button by pressing system back button </div><div>Device 1: Find `Button` by `xpath`: `//android.view.ViewGroup[contains(@content-desc,'Account 1')]` </div></blockquote></p>

```
critical/test_wallet.py:435: in test_wallet_bridge_flow_mainnet
    self.wallet_view.get_account_element().click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//android.view.ViewGroup[contains(@content-desc,'Account 1')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777479">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td><p><blockquote><div>Device 1: Swiping right on element SlideButton </div><div>Device 1: Find `SlideButton` by `xpath`: `//*[@resource-id='slide-button-track']` </div></blockquote></p>

```
critical/test_wallet.py:337: in test_wallet_send_flow_mainnet
    self.wallet_view.slide_button_track.slide()
../views/base_view.py:107: in slide
    self.swipe_right_on_element(width_percentage=1.3, start_x=100)
../views/base_element.py:288: in swipe_right_on_element
    location, size = self.get_element_coordinates()
../views/base_element.py:274: in get_element_coordinates
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: SlideButton by xpath: `//*[@resource-id='slide-button-track']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777474">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:246: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: zQ3...dWXh5 was not restored as a contact from waku backup!
E    Device 1: zQ3...Vacac was not restored as a contact from waku backup!
E    Device 1: admin_open was not restored from waku-backup!!
E    Device 1: member_open was not restored from waku-backup!!
E    Device 1: admin_closed was not restored from waku-backup!!
E    Device 1: member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESXdt4OQhIolFTOSLZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777471">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESXdt4OQhIolFTOSLZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777472">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESu3XasyFmZOY4Qsbe">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRw7uSmvr3WwWXtDw">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777476">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777475">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3777473">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESCD72TwxA7n3Dr5sU">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDi2SKikdQN9xB5Of">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/keycard/create/events.cljs
+++ b/src/status_im/contexts/keycard/create/events.cljs
@@ -20,6 +20,10 @@
   (rf/dispatch [:navigate-back])
   (rf/dispatch [:open-modal :screen/confirm-backup
                 {:masked-seed-phrase masked-seed-phrase
+                 :on-try-again       #(rf/dispatch [:open-modal :screen/backup-recovery-phrase-dark
+                                                    {:on-success         backup-recovery-phrase-success
+                                                     :masked-seed-phrase masked-seed-phrase
+                                                     :revealed?          true}])
                  :on-success         #(rf/dispatch [:keycard/create.phrase-backed-up
                                                     masked-seed-phrase])}]))
 

--- a/src/status_im/contexts/keycard/create/events.cljs
+++ b/src/status_im/contexts/keycard/create/events.cljs
@@ -22,7 +22,10 @@
                 {:masked-seed-phrase masked-seed-phrase
                  :on-try-again       #(rf/dispatch [:open-modal :screen/backup-recovery-phrase-dark
                                                     {:on-success         backup-recovery-phrase-success
-                                                     :masked-seed-phrase masked-seed-phrase
+                                                     :masked-seed-phrase (->> masked-seed-phrase
+                                                                              security/safe-unmask-data
+                                                                              (string/join " ")
+                                                                              security/mask-data)
                                                      :revealed?          true}])
                  :on-success         #(rf/dispatch [:keycard/create.phrase-backed-up
                                                     masked-seed-phrase])}]))

--- a/src/status_im/contexts/keycard/migrate/events.cljs
+++ b/src/status_im/contexts/keycard/migrate/events.cljs
@@ -57,6 +57,13 @@
   (rf/dispatch [:navigate-back])
   (rf/dispatch [:open-modal :screen/confirm-backup
                 {:masked-seed-phrase masked-seed-phrase
+                 :on-try-again       #(rf/dispatch [:open-modal :screen/backup-recovery-phrase-dark
+                                                    {:on-success         backup-recovery-phrase-success
+                                                     :masked-seed-phrase (->> masked-seed-phrase
+                                                                              security/safe-unmask-data
+                                                                              (string/join " ")
+                                                                              security/mask-data)
+                                                     :revealed?          true}])
                  :on-success         #(rf/dispatch [:keycard/migration.phrase-backed-up])}]))
 
 (rf/reg-event-fx :keycard/migration.get-phrase

--- a/src/status_im/contexts/profile/backup_recovery_phrase/view.cljs
+++ b/src/status_im/contexts/profile/backup_recovery_phrase/view.cljs
@@ -44,22 +44,23 @@
 
 (defn view
   []
-  (let [step-labels                             [:t/backup-step-1 :t/backup-step-2 :t/backup-step-3
-                                                 :t/backup-step-4]
-        checked?                                (reagent/atom
-                                                 {:0 false
-                                                  :1 false
-                                                  :2 false
-                                                  :3 false})
-        revealed?                               (reagent/atom false)
-        customization-color                     (rf/sub [:profile/customization-color])
-        {:keys [on-success masked-seed-phrase]} (rf/sub [:get-screen-params])
-        seed-phrase                             (reagent/atom (if masked-seed-phrase
-                                                                (->
-                                                                  (security/safe-unmask-data
-                                                                   masked-seed-phrase)
-                                                                  (string/split #"\s"))
-                                                                []))]
+  (let [step-labels         [:t/backup-step-1 :t/backup-step-2 :t/backup-step-3
+                             :t/backup-step-4]
+        checked?            (reagent/atom
+                             {:0 false
+                              :1 false
+                              :2 false
+                              :3 false})
+        {:keys [on-success masked-seed-phrase
+                revealed?]} (rf/sub [:get-screen-params])
+        revealed?           (reagent/atom revealed?)
+        customization-color (rf/sub [:profile/customization-color])
+        seed-phrase         (reagent/atom (if masked-seed-phrase
+                                            (->
+                                              (security/safe-unmask-data
+                                               masked-seed-phrase)
+                                              (string/split #"\s"))
+                                            []))]
     (fn []
       (let [theme (quo.theme/use-theme)]
         (rn/use-mount
@@ -108,8 +109,7 @@
             [quo/bottom-actions
              {:actions          :one-action
               :button-one-label (i18n/label :t/i-have-written)
-              :button-one-props {:disabled?           (some false? (vals @checked?))
-                                 :customization-color customization-color
+              :button-one-props {:customization-color customization-color
                                  :on-press            #(on-success (security/mask-data @seed-phrase))}}]
             [quo/text
              {:size  :paragraph-2

--- a/src/status_im/contexts/wallet/add_account/create_account/new_keypair/confirm_backup/view.cljs
+++ b/src/status_im/contexts/wallet/add_account/create_account/new_keypair/confirm_backup/view.cljs
@@ -31,7 +31,7 @@
     [(subvec result 0 2) (subvec result 2 4)]))
 
 (defn- cheat-warning
-  []
+  [on-try-again]
   (let [customization-color (rf/sub [:profile/customization-color])]
     [:<>
      [quo/drawer-top {:title (i18n/label :t/do-not-cheat)}]
@@ -42,6 +42,8 @@
       {:button-one-label (i18n/label :t/see-recovery-phrase-again)
        :button-one-props {:customization-color customization-color
                           :on-press            (fn []
+                                                 (when on-try-again
+                                                   (on-try-again))
                                                  (rf/dispatch [:hide-bottom-sheet])
                                                  (rf/dispatch [:navigate-back]))}}]]))
 
@@ -95,6 +97,7 @@
         incorrect-count              (reagent/atom 0)
         show-error?                  (reagent/atom false)
         {:keys [on-success
+                on-try-again
                 masked-seed-phrase]} (rf/sub [:get-screen-params])
         unmasked-seed-phrase         (security/safe-unmask-data masked-seed-phrase)
         random-phrase                (reagent/atom [])]
@@ -121,7 +124,9 @@
                                                 (do
                                                   (when (> @incorrect-count 0)
                                                     (rf/dispatch [:show-bottom-sheet
-                                                                  {:content cheat-warning}]))
+                                                                  {:content (fn []
+                                                                              [cheat-warning
+                                                                               on-try-again])}]))
                                                   (reset! incorrect-count (inc @incorrect-count))
                                                   (reset! show-error? true))))]
           [rn/view {:style {:flex 1}}

--- a/src/status_im/contexts/wallet/add_account/create_account/new_keypair/confirm_backup/view.cljs
+++ b/src/status_im/contexts/wallet/add_account/create_account/new_keypair/confirm_backup/view.cljs
@@ -42,10 +42,10 @@
       {:button-one-label (i18n/label :t/see-recovery-phrase-again)
        :button-one-props {:customization-color customization-color
                           :on-press            (fn []
-                                                 (when on-try-again
-                                                   (on-try-again))
                                                  (rf/dispatch [:hide-bottom-sheet])
-                                                 (rf/dispatch [:navigate-back]))}}]]))
+                                                 (rf/dispatch [:navigate-back])
+                                                 (when on-try-again
+                                                   (on-try-again)))}}]]))
 
 (defn- button
   [{:keys [word margin-right on-press]}]

